sky/utils/dag_utils.py

```diff
@@ -30,3 +33,17 @@ def dump_chain_dag_to_yaml(dag: dag_lib.Dag, path: str) -> None:
     for task in dag.tasks:
         configs.append(task.to_yaml_config())
     common_utils.dump_yaml(path, configs)
+
+
+def infer_and_fill_dag_name(dag: dag_lib.Dag) -> None:
+    # For a singleton task, override the dag name with task name, if it is
+    # not None. Otherwise, use the dag name.
```

Comment:
Can we have a docstr to clearly explain this does? Incl. the purpose (mostly for display in ...).

`_maybe_infer_and_fill_dag_and_task_names` seem more accurate, due to L49.

For L49: actually, do we need to fill in other non-named tasks?

Reply:
Good point! Moved the task name setting to this function.